src/appimagetool/appdirtool.go

```diff
@@ -1508,21 +1533,44 @@ func getQtPrfxpath(f *os.File, err error, qtVersion int) string {
 	// From the current location in the file, search to the next 0x00 byte
 	f.Seek(offset, 0)
 	length := ScanFile(f, search)
-	log.Println("Length of value of qt_prfxpath:", length)
-	// Now that we know where in the file the information is, go get it
-	f.Seek(offset, 0)
-	buf := make([]byte, length)
-	// Make a buffer that is exactly as long as the range we want to read
-	_, err = io.ReadFull(f, buf)
-	if err != nil {
-		helpers.PrintError("Unable to read qt_prfxpath", err)
-		os.Exit(1)
+
+	qt_prfxpath := ""
+	qtPrefixPathRequiresPatch := true
+
+	// When length is 0, Qt has been built as relocatable
+	if length == 0 {
+		qtPrefixPathRequiresPatch = false
+		// Directory should be in ../Qt${qtVersion}
+		// TODO: Check if this is true for relocatable binaries in Qt5
+		qt_prfxpath = filepath.Dir(library) + fmt.Sprintf("/../Qt%d", qtVersion)
```

Comment:
Not this case on Arch Linux.
I have `/usr/lib/libQt6Core.so` but directory on `/usr/lib/qt6`. And `qt_prfxpath` is really empty.

Reply:
Note: Before applying this patch, it cannot work on Arch Linux either. So it is not a regression.

Reply:
Okay, this may not be a problem because Arch's Qt is not relocatable, even though it doesn't have `qt_prfxpath`.

Reply:
Thanks for testing it, I'll see if I can fix it.